src/analyze/analyze-security.c

```diff
@@ -2760,19 +2759,15 @@ static int offline_security_checks(
 
                 k = manager_load_startable_unit_or_warn(m, NULL, prepared, &units[count]);
                 if (k < 0) {
-                        if (r == 0)
-                                r = k;
+                        RET_GATHER(r, k);
```

Comment:
You could also do this (though, strictly speaking not the same):
```c
if (RET_GATHER(r, k) < 0)
    continue;
```

Reply:
This would be quite different. We'd also reach `continue` if there was a *previous* failure, but that's not what we want.

Reply:
Right, nvm then.